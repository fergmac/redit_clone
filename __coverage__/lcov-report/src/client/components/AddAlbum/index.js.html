<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/client/components/AddAlbum/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/client/components/AddAlbum</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83</td><td class="line-coverage quiet"><span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { PureComponent } from 'r<span class="cstat-no" title="statement not covered" >eact';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
import styles from './style.css';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { saveAlbum } from '../../../shared/redux/modules/albums';
import Card from 'material-ui/Card';<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span></span></span></span></span></span></span></span></span>
// was create post
&nbsp;
class AddAl<span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >bu</span>m extends PureComponent {<span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" ><span class="fstat-no" title="function not covered" ></span></span></span></span></span></span></span>
&nbsp;
  componentDidMount() {
    th<span class="cstat-no" title="statement not covered" >is.props.saveAlbum({</span>
      title: 'ferg',
      description: 'fingerstyle guitar',
      link: 'https://www.youtube.com/watch?v=ygDs_pkEHDs',
    });
  }<span class="fstat-no" title="function not covered" ></span>
  render() {
    const { handleSubmit } = this.props;
    re<span class="cstat-no" title="statement not covered" >turn (</span>
      &lt;div className={styles.addAlbum}&gt;
        &lt;Card className={styles.addAlbumCard}&gt;
          &lt;h3&gt;Add Album Page&lt;/h3&gt;
          &lt;form onSubmit={handleSubmit}&gt;
            &lt;Field
              name="title"
              component="input"
              type="text"
              placeholder="Title"
            /&gt;
            &lt;Field
              name="artist"
              component="input"
              type="text"
              placeholder="Artist"
            /&gt;
            &lt;Field
              name="genre"
              component="input"
              type="text"
              placeholder="Genre"
            /&gt;
            &lt;Field
              name="description"
              component="input"
              type="text"
              placeholder="Description"
            /&gt;
            &lt;Field
              name="link"
              component="input"
              type="text"
              placeholder="Link"
            /&gt;
            &lt;Field
              name="tags"
              component="input"
              type="text"
              placeholder="Tags"
            /&gt;
            &lt;button type="submit"&gt;Submit&lt;/button&gt;
          &lt;/form&gt;
        &lt;/Card&gt;
      &lt;/div&gt;
      );
  }<span class="cstat-no" title="statement not covered" ></span>
}
&nbsp;
const mapStateToProps <span class="cstat-no" title="statement not covered" >= state =<span class="fstat-no" title="function not covered" >&gt; ({</span></span>
  albums: state.albums,
});
&nbsp;
const mapDispatchToProps <span class="cstat-no" title="statement not covered" >= (dispat<span class="fstat-no" title="function not covered" >ch) =&gt; ({</span></span>
  saveAlbum: (newAlbum) =&gt; {
    dispatch(saveAlbum(newAlbum));
  },
});
&nbsp;
export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'addalbum',
})(AddAlbum));
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Wed Nov 09 2016 16:25:23 GMT-0800 (PST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
