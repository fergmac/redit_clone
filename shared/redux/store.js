import { createStore } from 'redux';
import reducers from './reducers';

const store = createStore(reducers, {});

console.log(store);

export default store;
