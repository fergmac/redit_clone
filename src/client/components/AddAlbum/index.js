import React, { PureComponent } from 'react';
import styles from './style.css';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import { saveAlbum } from '../../../shared/redux/modules/albums';
import Card from 'material-ui/Card';
// was create post

class AddAlbum extends PureComponent {

  componentDidMount() {
    this.props.saveAlbum({
      title: 'ferg',
      description: 'fingerstyle guitar',
      link: 'https://www.youtube.com/watch?v=ygDs_pkEHDs',
    });
  }
  render() {
    const { handleSubmit } = this.props;
    return (
      <div className={styles.addAlbum}>
        <Card className={styles.addAlbumCard}>
          <h3>Add Album Page</h3>
          <form onSubmit={handleSubmit}>
            <Field
              name="title"
              component="input"
              type="text"
              placeholder="Title"
            />
            <Field
              name="artist"
              component="input"
              type="text"
              placeholder="Artist"
            />
            <Field
              name="genre"
              component="input"
              type="text"
              placeholder="Genre"
            />
            <Field
              name="description"
              component="input"
              type="text"
              placeholder="Description"
            />
            <Field
              name="link"
              component="input"
              type="text"
              placeholder="Link"
            />
            <Field
              name="tags"
              component="input"
              type="text"
              placeholder="Tags"
            />
            <button type="submit">Submit</button>
          </form>
        </Card>
      </div>
      );
  }
}

const mapStateToProps = state => ({
  albums: state.albums,
});

const mapDispatchToProps = (dispatch) => ({
  saveAlbum: (newAlbum) => {
    dispatch(saveAlbum(newAlbum));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm({
  form: 'addalbum',
})(AddAlbum));

