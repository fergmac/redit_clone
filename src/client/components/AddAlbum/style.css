.addAlbum {
  display: flex;

  width: 100%;
  height: 100%;
  flex-flow: column nowrap;
  align-items: center;
  justify-content: center;
}

.addAlbum form{
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
}