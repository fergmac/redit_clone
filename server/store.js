import { createStore } from 'redux';
import reducers from '../src/shared/redux/reducers';

export default createStore(reducers, {});


