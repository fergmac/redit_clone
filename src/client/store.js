import { createStore, applyMiddleware } from 'redux';
import createLogger from 'redux-logger';
import reduxThunk from 'redux-thunk';
import reducers from '../shared/redux/reducers';
import remoteMiddleware from './socket/remoteMiddleware';
import socket from './socket';

const logger = createLogger();

const middlewares = applyMiddleware(reduxThunk, remoteMiddleware(socket), logger);

const store = createStore(reducers, middlewares);

export default store;
