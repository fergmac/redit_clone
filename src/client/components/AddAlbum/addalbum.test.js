/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AddAlbum from './index';
import { Provider } from 'react-redux';
import store from '../../store';


describe('AddAlbum', () => {
    const albumData = {
        id: 1,
        title: 'Jazz Guitar',
        description: '1920s acoustic guitar jazz',
        votes: [{ id: 1 }]
    }
    it('creates a new Album', () => {
       const wrapper = shallow(
           <Provider store={store}>
                <AddAlbum 
                    album={albumData}
                />
          </Provider>
       )
        expect(shallowToJson(wrapper)).toMatchSnapshot()
    })
})