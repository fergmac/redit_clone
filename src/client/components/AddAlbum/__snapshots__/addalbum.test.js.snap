exports[`AddAlbum creates a new Album 1`] = `
<Connect(ReduxForm)
  album={
    Object {
      "description": "1920s acoustic guitar jazz",
      "id": 1,
      "title": "Jazz Guitar",
      "votes": Array [
        Object {
          "id": 1,
        },
      ],
    }
  } />
`;
